### DISCUSSION TITLE
Docs: Please add documentation on how to load server-side data for layouts and _app.js and _document.js

### DISCUSSION THREAD

**Discussion by Nefcanto** [vercel/next.js#34401] [Help] [answered]
### What is the improvement or update you wish to see?

A lot of websites have a layout that contains:

- Navigation
- Footer
- Sidebar
- ...

Sometimes these sections have something that is dynamic and needs to be loaded from the server.

In the documentation, it's [said](https://nextjs.org/docs/basic-features/layouts#data-fetching) that we should use client-side `useEffect` or `SWR` to load data for these sections.

But this approach is not good for SEO.

Please address this issue in the documentation. Please document how can we create SEO-friendly layouts, _app.js and _document.js.

Thank you.

### Is there any context that might help us understand?

We need to be able to somehow provide dynamic data for:

- Layouts
- _app.js
- _document.js

And if we can't do it, we need to have a place that explains it clearly and enumerates it as part of the problems of SEO with next.js.

### Does the docs page already exist? Please link to it.

https://nextjs.org/docs/basic-features/layouts#data-fetching

**Comment by Nefcanto** [comment]
No response for this yet? This is an extremely common use-case all across the web. We are a web-design agency.

Please provide us with a good solution regarding this issue.

Right now, we have to load layout data in all of our pages that use that data. This is extremely inefficient. It's like the worst part of next.js.

**Comment by Nefcanto** [comment]
Next.js, are you really going to not answer these important questions? At least not dropping a hint. Even after months?

**Comment by leerob** [comment, marked as answer]
**Update:** We’re excited to announce [Next.js 13](https://nextjs.org/13), which includes the new `app/` directory (beta) with support for nested layouts, colocation of data fetching, streaming, Server Components by default, and much more.

Inside the `app/` directory, there is a powerful new way to fetch data with React’s `use()` hook and the extended Web `fetch()` API. This allows you to colocate data fetching directly inside components for the most flexibility, including inside layouts.

You can leave feedback here: https://github.com/vercel/next.js/discussions/41745
